src/pages/commitqueue/CommitQueueCard.tsx

```diff
@@ -28,7 +35,26 @@ export const CommitQueueCard: React.FC<Props> = ({
   commitTime,
   patchId,
   moduleCodeChanges,
+  commitQueueId,
 }) => {
+  const [removePatchFromCommitQueue] = useMutation<
+    RemovePatchFromCommitQueueMutation,
+    RemovePatchFromCommitQueueMutationVariables
+  >(REMOVE_PATCH_FROM_COMMIT_QUEUE);
+  const [isButtonDisabled, setButtonDisabled] = useState(false);
+  const handleEnroll = async (e) => {
+    e.preventDefault();
+    setButtonDisabled(true);
+    try {
+      await removePatchFromCommitQueue({
+        variables: { patchId, commitQueueId },
+        refetchQueries: ["CommitQueue"],
+      });
+    } catch (error) {
+      setButtonDisabled(false);
```

Comment:
instead of using state, you could set button to disabled when `loading` from the mutation is `true`